test/integration_test/storkctl_schedulepolicy_test.go

```diff
@@ -0,0 +1,246 @@
+}
+
+func createCustomWeeklySchedulePolicyTest(t *testing.T) {
+	testrailID := 93193
+	args := map[string]string{
+		"day-of-week": "Friday",
+		"time":        "2:00AM",
+		"retain":      "3",
+	}
+	createSchedulePolicyTest(t, "Weekly", args, "custom-weekly-policy.yaml", testrailID)
+}
+
+func createCustomMonthlySchedulePolicyTest(t *testing.T) {
+	testrailID := 93194
+	args := map[string]string{
+		"date-of-month": "15",
+		"time":          "11:00AM",
+		"retain":        "3",
+	}
+	createSchedulePolicyTest(t, "Monthly", args, "custom-monthly-policy.yaml", testrailID)
+}
+
+func ValidateSchedulePolicyFromFile(t *testing.T, specFile string, policyName string) error {
+	if specFile == "" {
+		return fmt.Errorf("empty file path")
+	}
+	file, err := os.Open(specFile)
+	if err != nil {
+		return fmt.Errorf("error opening file %v: %v", specFile, err)
+	}
+	data, err := io.ReadAll(bufio.NewReader(file))
+	if err != nil {
+		return fmt.Errorf("error reading file %v: %v", specFile, err)
+	}
+	policy := &storkv1.SchedulePolicy{}
+	dec := yaml.NewYAMLOrJSONDecoder(bytes.NewReader([]byte(data)), len(data))
+	if err := dec.Decode(&policy); err != nil {
+		return err
+	}
+
+	if err := ValidateSchedulePolicy(t, policy, policyName); err != nil {
+		return err
+	}
+	return nil
+}
+
+func ValidateSchedulePolicy(t *testing.T, schedulePolicy *storkv1.SchedulePolicy, schedulePolicyName string) error {
+	//We want to validate if the created schedule policy resource matches our expectations
+	actualPolicy, err := storkops.Instance().GetSchedulePolicy(schedulePolicyName)
+	if err != nil {
+		logrus.Errorf("Unable to get the schedule policy")
+		return err
+	}
+	logrus.Info("Trying to validate if the created policy is per expectations")
+	//Validating schedulePolicy.Policy because schedulePolicy.metadata cannot be validated
+	require.Equal(t, schedulePolicy.Policy, actualPolicy.Policy, "Created schedule policy doesn't match the expected specification")
+	return nil
+}
+
+func SchedulePolicyCleanup(t *testing.T, policyName string) {
+	logrus.Info("Cleaning up created resources")
+	err := storkops.Instance().DeleteSchedulePolicy(policyName)
+	if err != nil {
+		logrus.Errorf("Unable to delete schedule policy %s", policyName)
+	}
+	waitInterval := 2 * time.Second
+	//check if the schedulePolicy is successfully deleted
+	f := func() (interface{}, bool, error) {
+		logrus.Infof("Checking if schedule policy resource is successfully deleted")
+		_, err := storkops.Instance().GetSchedulePolicy(policyName)
+		if err == nil {
+			return "", true, fmt.Errorf("get schedule policy: %s should have failed", policyName)
+		}
+		if !errors.IsNotFound(err) {
+			logrus.Infof("unexpected err: %v when checking deleted schedulePolicy: %s", err, policyName)
+			return "", true, err
+		}
+		//deletion done
+		logrus.Infof("schedule policy %s successfully deleted", policyName)
+		return "", false, nil
+	}
+	_, err = task.DoRetryWithTimeout(f, defaultWaitTimeout, waitInterval)
```

Comment:
nit.. no need of declaring the variable. We can just pass 2 * time.Second to the function.